x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_flyout/form_components/key_type.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { useValues, useActions } from 'kea';
+import { EuiFormRow, EuiSelect, EuiText, EuiLink } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { AppLogic } from '../../../../app_logic';
+import { CredentialsLogic } from '../../credentials_logic';
+import { TOKEN_TYPE_DESCRIPTION, TOKEN_TYPE_INFO, DOCS_HREF } from '../../constants';
+
+export const FormKeyType: React.FC = () => {
+  const { myRole } = useValues(AppLogic);
+  const { setTokenType } = useActions(CredentialsLogic);
+  const { activeApiToken, activeApiTokenExists } = useValues(CredentialsLogic);
+
+  const tokenDescription = TOKEN_TYPE_DESCRIPTION[activeApiToken.type];
+  const tokenOptions = TOKEN_TYPE_INFO.filter((typeInfo) =>
+    myRole?.credentialTypes?.includes(typeInfo.value)
+  );
+
+  return (
+    <EuiFormRow
+      label={i18n.translate('xpack.enterpriseSearch.appSearch.tokens.formType.label', {
+        defaultMessage: 'Key type',
+      })}
+      fullWidth
+      helpText={
+        <EuiText size="xs">
+          <p>
+            {tokenDescription && <strong>{tokenDescription}</strong>}{' '}
+            <EuiLink href={DOCS_HREF}>
+              {i18n.translate('xpack.enterpriseSearch.appSearch.tokens.documentationLink1', {
+                defaultMessage: 'Visit the documentation',
+              })}
+            </EuiLink>{' '}
+            {i18n.translate('xpack.enterpriseSearch.appSearch.tokens.documentationLink2', {
+              defaultMessage: 'to learn more about keys.',
+            })}
+          </p>
```

Comment:
I'm not sure if there's a better way of i18n'ing sentences broken up by JSX/HTML (a link in this case). Hoping our amazing i18n folks are ok handling this 😬